client/kafka/kafka.go

```diff
@@ -7,25 +7,27 @@ import (
 
 	"github.com/beatlabs/patron/correlation"
 	"github.com/beatlabs/patron/encoding"
+	"github.com/cloudevents/sdk-go/v2/binding"
 
 	"github.com/Shopify/sarama"
-	"github.com/opentracing/opentracing-go"
+	"github.com/cloudevents/sdk-go/protocol/kafka_sarama/v2"
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	opentracing "github.com/opentracing/opentracing-go"
 	"github.com/prometheus/client_golang/prometheus"
 )
 
 const (
-	asyncProducerComponent = "kafka-async-producer"
-	syncProducerComponent  = "kafka-sync-producer"
-	messageCreationErrors  = "creation-errors"
-	messageSendErrors      = "send-errors"
-	messageSent            = "sent"
+	messageCreationErrors = "creation-errors"
+	messageSendErrors     = "send-errors"
+	messageSent           = "sent"
 )
 
 var messageStatus *prometheus.CounterVec
 
 // Producer interface for Kafka.
```

Comment:
would it make sense to have a decorated producer, that sends always cloudevents in the 'send' method ?